### DISCUSSION TITLE
Express API: Cannot integrate CORS with Routers

### DISCUSSION THREAD

**Discussion by ceruleane** [vercel/vercel#9395] [Help]
Hello, I have made an Express API file (test.js) and loaded to Vercel following this tutorial: 

https://shadowsmith.com/how-to-deploy-an-express-api-to-vercel. 

Unfortunately, this uses "router", which is incompatible to "headers" that are necessary to avoid Access-Control-Allow-Domain issue according to Vercel guidelines. 

I have tried crudely including CORS in the express.js file:

```
require("dotenv").config({path:__dirname+'/.env'});

const cors = require('cors');
const express = require('express');
const bodyParser = require('body-parser');

const app = express();

// Serve the favicon from the public folder
app.use(express.static('public'));

app.use(cors({ origin: 'https://gptionary.com' }));
app.use(bodyParser.json());
```

but this doesn't work (i think). 

My current vercel.json is:

```
{
  "version": 2,
  "builds": [
    {
      "src": "test.js",
      "use": "@now/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "test.js"
    }
  ]
}
```

Is there a way to enable CORs and routers for the Express API file? Thank you so much for your interest and support :)

**Comment by Toadie11** [comment]
Hi man, I have the same issue. Did you already find a way to fix this?
